File: android/src/main/java/com/proventask/capacitor/GoogleOneTapAuth/SignInResult.java
package com.proventask.capacitor.GoogleOneTapAuth;

public class SignInResult {
    public String idToken;
    public String noSuccessReasonCode;
    public String noSuccessAdditionalInfo;

    public SignInResult(String noSuccessReasonCode, String noSuccessAdditionalInfo) {
        this.noSuccessReasonCode = noSuccessReasonCode;
        this.noSuccessAdditionalInfo = noSuccessAdditionalInfo;
    }

    public SignInResult(String idToken) {
        this.idToken = idToken;
    }
}

File: android/src/main/java/com/proventask/capacitor/googleOneTapAuth/SignInResult.kt
package com.proventask.capacitor.googleOneTapAuth

data class SignInResult(
    var idToken: String? = null,
    var noSuccessReasonCode: String? = null,
    var noSuccessAdditionalInfo: String? = null
) {
    constructor(noSuccessReasonCode: String?, noSuccessAdditionalInfo: String?) : this(
        null,
        noSuccessReasonCode,
        noSuccessAdditionalInfo
    )

    constructor(idToken: String?) : this(idToken, null, null)
}
